src/udev/udevd.c

```diff
@@ -65,70 +66,37 @@ static int listen_fds(int *ret_ctrl, int *ret_netlink) {
         return 0;
 }
 
+static DEFINE_CONFIG_PARSE_ENUM(config_parse_resolve_name_timing, resolve_name_timing, ResolveNameTiming, "Failed to parse resolve name timing");
+
 static int manager_parse_udev_config(Manager *manager) {
-        _cleanup_free_ char *log_val = NULL, *children_max = NULL, *exec_delay = NULL,
-                *event_timeout = NULL, *resolve_names = NULL, *timeout_signal = NULL;
+        int log_val = -1;
+        Manager *m = ASSERT_PTR(manager); /* A hack to check the ptr before using it in the table */
+        const ConfigTableItem config_table[] = {
+                { NULL, "udev_log",       config_parse_log_level,           0, &log_val },
+                { NULL, "children_max",   config_parse_unsigned,            0, &m->children_max },
+                { NULL, "exec_delay",     config_parse_sec,                 0, &m->exec_delay_usec },
+                { NULL, "event_timeout",  config_parse_sec,                 0, &m->timeout_usec },
+                { NULL, "resolve_names",  config_parse_resolve_name_timing, 0, &m->resolve_name_timing },
+                { NULL, "timeout_signal", config_parse_signal,              0, &m->timeout_signal },
+                {}
+        };
         int r;
 
-        assert(manager);
-
-        r = parse_env_file(NULL, "/etc/udev/udev.conf",
-                           "udev_log", &log_val,
-                           "children_max", &children_max,
-                           "exec_delay", &exec_delay,
-                           "event_timeout", &event_timeout,
-                           "resolve_names", &resolve_names,
-                           "timeout_signal", &timeout_signal);
+        r = config_parse_config_file_full(
+                        "udev.conf",
+                        "udev",
+                        NULL,
```

Comment:
as per current coding style, please prefix opaque parameters like this with their name, i.e. change this line to `/* sections= */ NULL,`